pkg/callback/callbacks.go

```diff
@@ -75,21 +75,16 @@ func (c Callback) ID() string {
 	return ""
 }
 
-// ThingIdentity allows a callback handler to request information from the Thing
-// This is especially important for dynamic data
-type ThingIdentity interface {
-	// Realm returns the realm that the identity belongs
-	Realm() string
-}
-
 // CallbackHandler is an interface for an AM callback handler
 type Handler interface {
 	// Handle the callback by modifying it
-	Handle(id ThingIdentity, cb Callback, metadata *AuthMetadata) error
+	Handle(cb Callback) error
 }
 
-type AuthMetadata struct {
-	ConfirmationKey crypto.Signer
+// RestrictedHandler responds to an AM callback that creates a restricted session
+type RestrictedHandler interface {
```

Comment:
I think it might be more accurate if this is call `ProofOfPossessionHandler` with a `SigningKey` function:
```
type ProofOfPossessionHandler interface {
	Handler
	SigningKey() crypto.Signer
}
```

I think it's better to use "confirmation key" to refer to the public key that AM uses to do the verification.

Reply:
@JacoJooste - I suppose it makes sense to rename the `ConfirmationKey` method on the Session struct as well? E.g.

```
func (s *Session) ConfirmationKey() crypto.Signer
```
to
```
func (s *Session) SigningKey() crypto.Signer
```

Reply:
Yes. That does make sense 👍